samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,694 @@
+		      &gen_onoff_pub_srv_s_1, &onoff_state[2]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli_s_1, &onoff_state[2]),
+};
+
+/*
+ * Element 3 Models
+ */
+
+static struct bt_mesh_model secondary_2_models[] = {
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op,
+		      &gen_onoff_pub_srv_s_2, &onoff_state[3]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli_s_2, &onoff_state[3]),
+};
+
+/*
+ * Button to Client Model Assignments
+ */
+
+struct bt_mesh_model *mod_cli_sw[] = {
+		&root_models[4],
+		&secondary_0_models[1],
+		&secondary_1_models[1],
+		&secondary_2_models[1],
+};
+
+/*
+ * LED to Server Model Assigmnents
+ */
+
+struct bt_mesh_model *mod_srv_sw[] = {
+		&root_models[3],
+		&secondary_0_models[0],
+		&secondary_1_models[0],
+		&secondary_2_models[0],
+};
+
+/*
+ * Root and Secondary Element Declarations
+ */
+
+static struct bt_mesh_elem elements[] = {
+	BT_MESH_ELEM(0, root_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_0_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_1_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_2_models, BT_MESH_MODEL_NONE),
+};
+
+static const struct bt_mesh_comp comp = {
+	.cid = CID_INTEL,
+	.elem = elements,
+	.elem_count = ARRAY_SIZE(elements),
+};
+
+struct device *sw_device;
+
+struct sw {
+	u32_t pin;
+	struct k_work button_work;
+};
+
+static struct sw sw;
+
+static struct gpio_callback button_cb;
+
+static u8_t trans_id;
+static u32_t time, last_time;
+static u16_t primary_addr;
+static u16_t primary_net_idx;
+
+/*
+ * Generic OnOff Model Server Message Handlers
+ *
+ * Mesh Model Specification 3.1.1
+ *
+ */
+
+static void gen_onoff_get(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(2 + 1 + 4);
+	struct onoff_state *onoff_state = model->user_data;
+
+	printk("%s addr: 0x%04x onoff: 0x%02x\n"
+			, __func__, model->elem->addr, onoff_state->current);
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_GEN_ONOFF_STATUS);
+	net_buf_simple_add_u8(msg, onoff_state->current);
+
+	if (bt_mesh_model_send(model, ctx, msg, NULL, NULL)) {
+		printk("%s Unable to send On Off Status response\n",
+				__func__);
+	}
+}
+
+static void gen_onoff_set_unack(struct bt_mesh_model *model,
+				struct bt_mesh_msg_ctx *ctx,
+				struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = model->pub->msg;
+	struct onoff_state *onoff_state = model->user_data;
+	int err;
+	u8_t local_trans_id;
+
+	/* Ignore command sent by us to us as the LED state is set by button */
+	if (model->elem->addr != ctx->addr) {
+		onoff_state->current = net_buf_simple_pull_u8(buf);
+		local_trans_id = net_buf_simple_pull_u8(buf);
+		printk("%s addr: 0x%02x state: 0x%02x trans_id:"
+				" 0x%02x src: 0x%04x\n",
+				__func__, model->elem->addr,
+				onoff_state->current,
+				local_trans_id, ctx->addr);
```

Comment:
When you split a function call or function definition into multiple lines, the split lines should be aligned so that they start right after the ( on the first line. I.e. tabs + 0-7 spaces. I've noted this in a few other places but not all of them, since I guess you get the idea. Another thing to avoid is splitting string literal across lines since it hurts grepping for them based on output that you see on the console, however I think it might not be possible in this case. One thing that'd shorted the string is to drop the ':', at least that's the convention I've used when doing logging in the mesh code.

Reply:
ok